app/backends/workflow-orchestration.ts

```diff
@@ -526,138 +537,162 @@ export async function handleWorkItemUpdate(
-      duration,
-      totalItemsSize,
-      outputItemSizes);
-
-    const completedWorkItemCount = await workItemCountForStep(
-      tx, workItem.jobID, workItem.workflowStepIndex, COMPLETED_WORK_ITEM_STATUSES,
-    );
-    const allWorkItemsForStepComplete = (completedWorkItemCount == thisStep.workItemCount);
-
-    if (hits && job.numInputGranules > hits) {
-      job.numInputGranules = hits;
-      await job.save(tx);
-      await updateWorkItemCounts(tx, job);
-    }
+      if (!totalItemsSize && outputItemSizes?.length > 0) {
+        totalItemsSize = sum(outputItemSizes) / 1024 / 1024;
+      }
 
-    const continueProcessing = await handleFailedWorkItems(tx, job, workItem, thisStep, status, logger, errorMessage);
-    if (continueProcessing) {
-      let nextStep = null;
-      if (status != WorkItemStatus.FAILED) {
-        nextStep = await createNextWorkItems(
-          tx,
-          logger,
-          workItem,
-          allWorkItemsForStepComplete,
-          results,
-          outputItemSizes);
+      await updateWorkItemStatus(
+        tx,
+        workItemID,
+        status as WorkItemStatus,
+        duration,
+        totalItemsSize,
+        outputItemSizes);
+
+      const completedWorkItemCount = await workItemCountForStep(
+        tx, workItem.jobID, workItem.workflowStepIndex, COMPLETED_WORK_ITEM_STATUSES,
+      );
+      const allWorkItemsForStepComplete = (completedWorkItemCount == thisStep.workItemCount);
+
+      if (hits && job.numInputGranules > hits) {
+        job.numInputGranules = hits;
+        await job.save(tx);
+        await updateWorkItemCounts(tx, job);
       }
 
-      if (nextStep) {
-        if (results && results.length > 0) {
-          // set the scrollID for the next work item to the one we received from the update
-          workItem.scrollID = scrollID;
-          await maybeQueueQueryCmrWorkItem(tx, workItem, logger);
-        } else {
-          // Failed to create the next work items - fail the job rather than leaving it orphaned
-          // in the running state
-          logger.error('The work item update should have contained results to queue a next work item, but it did not.');
-          const message = 'Harmony internal failure: could not create the next work items for the request.';
-          await completeJob(tx, job, JobStatus.FAILED, logger, message);
-        }
-      } else {
-        // Finished with the chain for this granule
-        if (status != WorkItemStatus.FAILED) {
-          await addJobLinksForFinishedWorkItem(tx, job, results, logger);
+      const continueProcessing = await handleFailedWorkItems(tx, job, workItem, thisStep, status, logger, errorMessage);
+      if (continueProcessing) {
+        const nextWorkflowStep = await getWorkflowStepByJobIdStepIndex(
+          tx, workItem.jobID, workItem.workflowStepIndex + 1,
+        );
+        if (nextWorkflowStep && (status !== WorkItemStatus.FAILED || nextWorkflowStep?.isBatched)) {
```

Comment:
The ignoreErrors logic is handled entirely within `handleFailedWorkItems` (one of the few places we've managed to do that in workflow orchestration). It will not enter this block if `ignoreErrors` is false and the status is failed because `continueProcessing` will be false if there's an error.